src/problem/MFEMProblem.C

```diff
@@ -259,17 +263,31 @@ MFEMProblem::addKernel(const std::string & kernel_name,
     mfem_problem_builder->AddKernel<mfem::ParBilinearForm>(parameters.get<std::string>("variable"),
                                                            blf_kernel->getKernel());
   }
+  else
+  {
+    mooseError("Unsupported kernel of type '", kernel_name, "' and name '", name, "' detected.");
+  }
 }
 
 void
 MFEMProblem::addAuxKernel(const std::string & kernel_name,
                           const std::string & name,
                           InputParameters & parameters)
 {
-  FEProblemBase::addUserObject(kernel_name, name, parameters);
-  MFEMAuxSolver * mfem_auxsolver(&getUserObject<MFEMAuxSolver>(name));
-  mfem_problem_builder->AddPostprocessor(name, mfem_auxsolver->getAuxSolver(), true);
-  mfem_auxsolver->storeCoefficients(_coefficients);
+  // NB: - slightly crude test.
+  bool is_mfem_auxkernel = (strncmp(kernel_name.c_str(), "MFEM", 4) == 0);
```

Comment:
This is indeed fragile - should have a check on the base class, at least.

Reply:
This has been resolved in the latest commit by registering MFEMAuxKernel as a base class. Now in the MFEMProblem addAuxKernel method, we can determine whether the auxkernel is an MFEM auxsolver or a MOOSE auxkernel.